File: leetcode_projects/c++_codes/ex1Q.cpp
#include <iostream>
#include <vector>

std::vector<int> twoSum(std::vector<int>& nums, int target){
    for(int i = 0; i < nums.size() - 1; i++){
        for(int j = 0; j < nums.size(); j++){
            if(nums[i] + nums[j] == target)//se alvo encontrado retorno as posições dos itens que somados chegam ao alvo
                return {i, j};
        }
    }

    return {-1, -1};
}

int main(){
    std::vector<int> nums = {2, 7, 11, 15};
    int target = 9;

    std::vector<int> result = twoSum(nums, target);

    std::cout << "índices: [" << result[0] << ", " << result[1] << "]\n";

    return 0;
}

File: leetcode_projects/go_codes/ex1Q.go
package main

import "fmt"
//função que acha os dois valores no array que somados são iguais ao alvo
//implementação O(n²)
func twoSum(nums []int, target int) []int{
    for i := 0; i < len(nums) - 1; i++{
        for j := i + 1; j < len(nums); j++{
            if nums[i] + nums[j] == target{
                return []int{i, j}
            }
        }
    }

    return []int{-1, -1}
}

func main(){
    nums := []int{2, 7, 11, 15}
    target := 18

    fmt.Println(twoSum(nums, target))
}
